app/src/main/java/ru/practicum/android/diploma/filters/data/repository/FilterRepositoryImpl.kt

```diff
@@ -0,0 +1,79 @@
+package ru.practicum.android.diploma.filters.data.repository
+
+import android.content.SharedPreferences
+import ru.practicum.android.diploma.filters.domain.api.FilterRepository
+
+class FilterRepositoryImpl(
+    private val sharedPrefs: SharedPreferences,
+) : FilterRepository {
+    override fun saveFilters(
+        location: String?,
+        industry: String?,
+        expectedSalary: String?,
+        removeNoSalary: Boolean,
+        areaId: String?,
+    ) {
+        var isFiltersOn = false
+        val country: String?
+        var region: String? = null
+        if (location.toString().contains(",")) {
+            country = location?.split(",")?.first()
+            region = location?.split(",")?.last()
+        } else {
+            country = location
+        }
+        if (!areaId.isNullOrEmpty() || !industry.isNullOrEmpty() || !expectedSalary.isNullOrEmpty() || removeNoSalary) {
+            isFiltersOn = true
+        }
+
+        sharedPrefs.edit()
+            .putString(COUNTRY_KEY, country)
+            .putString(REGION_KEY, region)
+            .putString(INDUSTRY_KEY, industry)
+            .putString(EXPECTED_SALARY_KEY, expectedSalary)
+            .putBoolean(NO_SALARY_KEY, removeNoSalary)
+            .putString(AREA_ID_KEY, areaId)
+            .putBoolean(FILTERS_ON_KEY, isFiltersOn)
+            .apply()
+    }
+
+    override fun getLocation(): String {
+        val location: String = if (sharedPrefs.getString(REGION_KEY, null) !== null) {
+            sharedPrefs.getString(COUNTRY_KEY, null) + "," + sharedPrefs.getString(REGION_KEY, null)
+        } else {
+            sharedPrefs.getString(COUNTRY_KEY, null) ?: ""
+        }
+        return location
```

Comment:
Здесь удобно было бы применить StringBuilder:
```suggestion
       return buildString { 
            append(sharedPrefs.getString(COUNTRY_KEY, null) ?: "")
            sharedPrefs.getString(REGION_KEY, null)?.let {
                append(",$it")
            }
        }
```